test/conv.jl

```diff
@@ -725,6 +725,17 @@ end
     @test size(conv(x, w; stride = (1, 2), pad = (2, 3), dilation = (2, 2), flipped = true)) == (12, 7, 16, 10)
 end
 
+# https://github.com/FluxML/NNlib.jl/issues/369
+@testset "conv_wrapper with groups - not equal types that trigger direct backend" begin
+    x = rand(Float32, 10, 10, 32, 8)
+    w = rand(Float64, 2, 2, 16, 4)
+    g = 2
+    @test size(conv(x, w; groups=g)) == (9, 9, 4, 8)
```

Comment:
Perhaps better to test that the values are the same as the other implementation; this will also test the sizes:
```suggestion
    @test conv(x, w; groups=g) ≈ conv(x, Float32.(w); groups=g)
```

Reply:
Updated this to all tests